### PR TITLE
Feature/state_improvement

### PR DESCRIPTION
**Pull request by TimonKapischke**
- coding style
- Makefiles with common.mk
- update State.h, State.cc
- update "tests"

### CODE REVIEW

File: src/alignment/State.cc [hunk 1]
```diff
@@ -60,67 +64,62 @@ void state::updateSelectability(int i){
     }
 } 
 
+bool state::consistent(Edge& e, Edge& f){
+   if ((e.first->j <= f.first->j && e.second->j >= f.second->j)
```
Comment (MarioStanke) [inline src/alignment/State.cc]:
```suggestion
   // requires that e.first->i = f.first->i and e.second->i = f.second->i
   if ((e.first->j <= f.first->j && e.second->j >= f.second->j)
```

File: src/alignment/State.cc [hunk 2]
```diff
@@ -60,67 +64,62 @@ void state::updateSelectability(int i){
...
+       || (e.first->j >= f.first->j && e.second->j <= f.second->j)) {
+       return false;
+   } else if (e.first->j == f.first->j && e.second->j == f.second->j) {
```
Comment (MarioStanke) [inline src/alignment/State.cc]: This is a bug. This condition to check whether e and f are the same edges should be the first condition. As it is, this condition is never true. Please, also give a short comment on this condition, like `// edges e and f are equal`.

Comment (TimonKapischke) [reply src/alignment/State.cc]: done

File: src/alignment/State.cc (hunk 1 above)
Comment (TimonKapischke) [reply src/alignment/State.cc]: done

**Pull request merged**